### PR TITLE
feat(persons): Allow selecting IDs to see more accurate feature flag …

### PR DESCRIPTION
**Pull request by neilkakkar**
…values

image

image

image

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by lharries** [comment]
Instead of an ID selector would it be possible to allow them to fuzzy enter the name? People likely remember the name and would otherwise need to hunt around for the ID then copy and paste

(Fully up to you and feel free to add as a separate PR if you prefer)

File: frontend/src/scenes/persons/Person.tsx [hunk 1]
```diff
@@ -201,7 +207,25 @@ export function Person(): JSX.Element | null {
...
-                        <RelatedFeatureFlags distinctId={person.distinct_ids[0]} />
+                        <div className="flex space-x-4 items-center mb-2">
+                            <div>
+                                Choose ID:
+                                <Tooltip title="Feature flags can have different values based on the persons IDs. Turn on persistence in feature flag settings if you'd like these to be constant always.">
+                                    <InfoCircleOutlined style={{ marginLeft: 6, marginRight: 0 }} />
+                                </Tooltip>
+                            </div>
+                            <LemonSelect
+                                value={person.distinct_ids[0]}
```
Comment (EDsCODE) [inline frontend/src/scenes/persons/Person.tsx:218]: The force index here is 😬 but i guess it should never be empty

**Review by EDsCODE — APPROVED** [review]
Love this small change huge improvement

**Comment by neilkakkar** [comment]
@lharries this happens _after_ you've already found the person you're searching for. These IDs would basically be the list of IDs you've called `identify` with, prepopulated here, no search required.

File: frontend/src/scenes/persons/Person.tsx (hunk 1 above)
Comment (neilkakkar) [reply frontend/src/scenes/persons/Person.tsx:218]: Yeah, the person can't exist without atleast one distinct ID.

**Comment by neilkakkar** [comment]
oh wait, you mean search on feature flag? That's already fuzzy 🤔 . Think it's not clear from the screenshots, let me merge, keen to hear your feedback once you see it in prod 😂

**Comment by lharries** [comment]
@neilkakkar realized that after I commented

Out of interest why do we allow 1 person to have multiple distinct IDs each with different feature flags rather than making the feature flags all the same for the 1 ID?

I've suggested a small wording change to make it a bit clearer

**Comment by neilkakkar** [comment]
@lharries you'll enjoy this: https://github.com/PostHog/posthog/issues/9547#issuecomment-1148612360

🙈  - small reason is that person IDs can easily be inconsistent (they can change when users are merged for example), and don't allow us to get all the perf improvements we did get with local evaluation / bootstrapping, since personIDs are our backend identifiers, which users aren't aware of.

**Comment by lharries** [comment]
@neilkakkar I suspected something like that!

**Pull request merged**